src/_P003_Pulse.ino

```diff
@@ -4,56 +4,106 @@
 //
 // Make sure physical connections are electrically well sepparated so no crossover of the signals happen.
 // Especially at rates above ~5'000 RPM with longer lines. Best use a cable with ground and signal twisted.
+// The Mode Types "PULSE low/high/change" are suited for low frequence pulses but for and precise counting 
+// with pulse rates of less than 750 RPM with DebounceTime > 20ms and pulse length > 40ms. This type may
+// tolerate less good signals. After a pulse and debounce time it verifies the signal 3 times.
 
 #include "src/Helpers/ESPEasy_time_calc.h"
 
-#define PLUGIN_003
-#define PLUGIN_ID_003         3
-#define PLUGIN_NAME_003       "Generic - Pulse counter"
-#define PLUGIN_VALUENAME1_003 "Count"
-#define PLUGIN_VALUENAME2_003 "Total"
-#define PLUGIN_VALUENAME3_003 "Time"
+// additional debug/tuning messages from PULSE mode into the logfile
+#define P003_PULSE_STATISTIC
+#define P003_PULSE_STATS_LOG_LEVEL     LOG_LEVEL_DEBUG
 
+#define PLUGIN_003
+#define PLUGIN_ID_003                  3
+#define PLUGIN_NAME_003                "Generic - Pulse counter"
+// number and name of counted values 
+#define PLUGIN_NR_VALUENAMES_003       3
+#define PLUGIN_VALUENAME1_003          "Count"
+#define PLUGIN_VALUENAME2_003          "Total"
+#define PLUGIN_VALUENAME3_003          "Time"
+// ... their index in UserVar and TaskDeviceValueNames
+#define P003_IDX_pulseCounter           0
+#define P003_IDX_pulseTotalCounter      1
+#define P003_IDX_pulseTime              2
+// ... and the following index into UserVar for storing the persisted TotalCounter
+#define P003_IDX_persistedTotalCounter  3
+
+// indexes for config parameters
+#define P003_IDX_DEBOUNCETIME   0 
+#define P003_IDX_COUNTERTYPE    1
+#define P003_IDX_MODETYPE       2
+// special Mode Type. Note: Lower 3 bits are significant for GPIO Interupt type. The upper bits distinguish the Mode Types
+#define PULSE_LOW               (0x10|CHANGE)
+#define PULSE_HIGH              (0x20|CHANGE)
+#define PULSE_CHANGE            (0x30|CHANGE)
+#define PULSE_MODE_MASK         0x30
+#define MODE_INTERRUPT_MASK     0x03
+
+// values for WEBFORM Counter Types
+#define P003_NR_COUNTERTYPES               4
+#define P003_COUNTERTYPE_LIST { F("Delta"), F("Delta/Total/Time"), F("Total"), F("Delta/Total"), }
+#define P003_CT_INDEX_COUNTER              0
+#define P003_CT_INDEX_COUNTER_TOTAL_TIME   1
+#define P003_CT_INDEX_TOTAL                2
+#define P003_CT_INDEX_COUNTER_TOTAL        3
+// processing Steps in PLUGIN_TIMER_IN
+#define P003_PROCESSING_STEP_1             1
+#define P003_PROCESSING_STEP_2             2
+#define P003_PROCESSING_STEP_3             3
 
 void Plugin_003_pulse_interrupt1() ICACHE_RAM_ATTR;
 void Plugin_003_pulse_interrupt2() ICACHE_RAM_ATTR;
 void Plugin_003_pulse_interrupt3() ICACHE_RAM_ATTR;
 void Plugin_003_pulse_interrupt4() ICACHE_RAM_ATTR;
-void Plugin_003_pulsecheck(byte Index) ICACHE_RAM_ATTR;
+void Plugin_003_pulsecheck(byte taskID) ICACHE_RAM_ATTR;
```

Comment:
It is a `taskIndex_t`, so why not use that typedef?

Reply:
Hmm, simply because this is part of the things that I did not change in the original code. But I will have a look, because it applies to various declarations in the code.

Reply:
> This will create the documentation in HTML format in a folder where you have your repository cloned.
> For me it is: `C:\GitHub\TD-er\ESPEasy\docs\build\html` Then open `index.html` in a browser to see your result.

Ok. Thank you. That did work.
I also tried some rst-Viewers in VSC, but they did either not work or had poor results. So I had to go the long way.

Documentation is now ready in ESPEasy/docs/source/Plugin/. So You can pull it.

Reply:
> It is a `taskIndex_t`, so why not use that typedef?

Ok, I changed it.
So I am now ready.